packages/grid/x-data-grid/src/hooks/features/editRows/useGridCellEditing.new.ts

```diff
@@ -130,6 +130,11 @@ export const useGridCellEditing = (
         if (event.key === 'Escape') {
           reason = GridCellEditStopReasons.escapeKeyDown;
         } else if (event.key === 'Enter') {
+          // Wait until IME is settled for Asian languages like Japanese and Chinese
+          // TODO: `event.which` is depricated but this is a temporary workaround
+          if (event.which === 229) {
```

Comment:
Let's to this for old editing implementation as well, since it's an easy change: https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid/src/hooks/features/editRows/useGridCellEditing.old.ts